source/AI.cpp

```diff
@@ -309,6 +312,199 @@ AI::AI(const List<Ship> &ships, const List<Minable> &minables, const List<Flotsa
 
 
 // Fleet commands from the player.
+void AI::IssueFormationChange(const PlayerInfo &player)
+{
+	// Figure out what ships we are giving orders to
+	vector<Ship *> ships;
+	bool fullFleet = player.SelectedShips().empty();
+	if(fullFleet)
+	{
+		for(const shared_ptr<Ship> &it : player.Ships())
```

Comment:
I'll reserve `player.Ships().size() - 1` for the full fleet case and `selectedShips.size()` for the case when there are selected ships. This would potentially reserve more space than we finally need (due to destroyed ships and parked ships), but I expect that performance would still be better than without reserve.